jena-arq/src/main/java/org/apache/jena/sparql/expr/aggregate/AggMedianDistinct.java

```diff
@@ -0,0 +1,119 @@
+import org.apache.jena.sparql.expr.ExprEvalException ;
+import org.apache.jena.sparql.expr.ExprList ;
+import org.apache.jena.sparql.expr.NodeValue ;
+import org.apache.jena.sparql.expr.nodevalue.XSDFuncOp ;
+import org.apache.jena.sparql.function.FunctionEnv ;
+
+public class AggMedianDistinct extends AggregatorBase
+{
+    // ---- Median(DISTINCT expr)
+    public AggMedianDistinct(Expr expr) { super("Median", true, expr) ; } 
+    @Override
+    public Aggregator copy(ExprList expr) { return new AggMedianDistinct(expr.get(0)) ; }
+
+    private static final NodeValue noValuesToMedian = NodeValue.nvZERO ; 
+
+    @Override
+    public Accumulator createAccumulator()
+    { 
+        return new AccMedianDistinct(getExpr()) ;
+    }
+
+    @Override
+    public Node getValueEmpty()     { return NodeValue.toNode(noValuesToMedian) ; } 
+
+    @Override
+    public int hashCode()   {
+        return HC_AggMedianDistinct ^ getExprList().hashCode() ;
+    }
+
+    @Override
+    public boolean equals(Aggregator other, boolean bySyntax) {
+        if ( other == null ) return false ;
+        if ( this == other ) return true ;
+        if ( ! ( other instanceof AggMedianDistinct ) ) return false ;
+        AggMedianDistinct a = (AggMedianDistinct)other ;
+        return exprList.equals(a.exprList, bySyntax) ;
+    }
+
+    
+    // ---- Accumulator
+    class AccMedianDistinct extends AccumulatorExpr
+    {
+        // Non-empty case but still can be nothing because the expression may be undefined.
+        private NodeValue total = noValuesToMedian ;
+        private int count = 0 ;
+        ArrayList<NodeValue> collection=new ArrayList<NodeValue>(); 
+        
+        static final boolean DEBUG = false ;
+        
+        public AccMedianDistinct(Expr expr) { super(expr, true) ; }
+
+        @Override
+        protected void accumulate(NodeValue nv, Binding binding, FunctionEnv functionEnv)
+        { 
+            if ( nv.isNumber() )
+            {
+                count++ ;
+                if ( total == noValuesToMedian ) {
+                    total = nv ;
+                }
+                else {
+                    total = XSDFuncOp.numAdd(nv, total) ;
+                }
+            	
+                collection.add(nv);
+            }
+            else
+                throw new ExprEvalException("median: not a number: "+nv) ;
+
+            if ( DEBUG ) System.out.println("median: ("+total+","+count+")") ;
+        }
+
+        @Override
+        public NodeValue getAccValue()
+        {
+            if ( count == 0 ) return noValuesToMedian ;
+            if ( super.errorCount != 0 )
+                return null ;
+            
+            double[] arrDouble = new double[collection.size()];
+            for(int i=0; i<collection.size(); i++){
+            	arrDouble[i] = collection.get(i).getDouble();            	
+            }
+            System.out.println(new Median().evaluate(arrDouble));
```

Comment:
Unprotected `System.out.println`.

Reply:
certainly a genuine mistake here. System.out.println and quick tests should have no place here.